packages/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerModule.java

```diff
@@ -663,12 +663,13 @@ private String writeVideo(Uri uri) {
     try {
       in = getApplication(mPromise).getContentResolver().openInputStream(uri);
       if (in != null) {
-        byte[] buffer = new byte[in.available()];
-        in.read(buffer);
-
         path = ExpFileUtils.generateOutputPath(mContext.getCacheDir(), "ImagePicker", ".mp4");
         out = new FileOutputStream(path);
-        out.write(buffer);
+        byte[] buffer = new byte[4096];
+        int bytesRead;
+        while ((bytesRead = in.read(buffer)) > 0) {
```

Comment:
This condition seems wrong.
"the total number of bytes read into the buffer, or -1 if there is no more data because the end of the file has been reached."

Should it be? `while ((bytesRead = in.read(buffer)) != -1) {`

Reply:
Apparently there is built-in-java-sdk function `Files.copy` (http://hg.openjdk.java.net/jdk8u/jdk8u/jdk/file/be30cb2a3088/src/share/classes/java/nio/file/Files.java#l2902) that performs copying exactly the same way 😄 
But thank you for pointing that out, because I will replace my implementation with using already-provided one 🙇

Reply:
It turned out that using `Files.copy` would be harder than just invoking that method, because it requires some specific arguments that are not easily obtainable - therefore I leave the code as is.